firmware/controllers/engine_cycle/high_pressure_fuel_pump.cpp

```diff
@@ -12,12 +12,170 @@
 
 #include "high_pressure_fuel_pump.h"
 #include "spark_logic.h"
+#include "fuel_computer.h"
 
 #if EFI_HPFP
 
+class HpfpActor {
+public:
+	DECLARE_ENGINE_PTR;
+	
+	void start() {
+		INJECT_ENGINE_REFERENCE(&m_lobe);
+		scheduleNextCycle();
+	}
+
+private:
+	AngleBasedEvent m_open;
+	AngleBasedEvent m_close;
+
+	HpfpLobe     m_lobe;
+	HpfpQuantity m_quantity;
+
+	void scheduleNextCycle();
+
+	static void hpfpPinTurnOn(HpfpActor *) {
+		enginePins.hpfpValve.setHigh();
+	}
+
+	static void hpfpPinTurnOff(HpfpActor *current) {
+		enginePins.hpfpValve.setLow();
+
+		current->scheduleNextCycle();
+	}
+};
+
+static HpfpActor actor; // TODO: Instantiate one per bank? Need more sensors....
```

Comment:
move into Engine? at least one benefit would be not sharing instance between unit tests, i.e. no state leak between tests